director/accounts/views.py

```diff
@@ -73,16 +76,34 @@ def has_any_permissions(
 
         return False
 
+    def is_permitted(
```

Comment:
Here is the `is_permitted` method as requested, however I don't think it will be called directly much, instead see the `request_permissions_guard` below.